src/main/java/com/mraof/minestuck/world/gen/feature/structure/FrogTemplePiece.java

```diff
@@ -0,0 +1,291 @@
+package com.mraof.minestuck.world.gen.feature.structure;
+
+import com.mraof.minestuck.block.DecorBlock;
+import com.mraof.minestuck.block.LotusTimeCapsuleBlock;
+import com.mraof.minestuck.block.MSBlocks;
+import com.mraof.minestuck.block.MSDirectionalBlock;
+import com.mraof.minestuck.entity.MSEntityTypes;
+import com.mraof.minestuck.entity.LotusFlowerEntity;
+import com.mraof.minestuck.world.gen.feature.MSStructurePieces;
+import com.mraof.minestuck.world.gen.feature.structure.blocks.StructureBlockUtil;
+import com.mraof.minestuck.world.storage.loot.MSLootTables;
+import net.minecraft.block.*;
+import net.minecraft.fluid.Fluids;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.state.properties.BlockStateProperties;
+import net.minecraft.state.properties.ChestType;
+import net.minecraft.state.properties.Half;
+import net.minecraft.util.Direction;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.ChunkPos;
+import net.minecraft.util.math.MutableBoundingBox;
+import net.minecraft.world.IWorld;
+import net.minecraft.world.gen.ChunkGenerator;
+import net.minecraft.world.gen.Heightmap;
+import net.minecraft.world.gen.feature.structure.ScatteredStructurePiece;
+import net.minecraft.world.gen.feature.structure.StructurePiece;
+import net.minecraft.world.gen.feature.template.TemplateManager;
+
+import java.util.Random;
+
+public class FrogTemplePiece extends ScatteredStructurePiece
+{
+	private boolean createRan = false; //boolean check to prevent certain objects(the lotus flower entity) from spawning several times over
```

Comment:
This variable should probably be read from/written to nbt

Reply:
Done!